src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -1196,5 +1199,224 @@ internal static bool TryGetInt(FormulaValue value, out int outputValue)
+                }
+
+                string json = JsonInternal(_args[0], 0);
+
+                if (json == null)
+                {
+                    return FormulaValue.NewBlank();
+                }
+
+                return new StringValue(_irContext, json);
+            }
+
+            private void GetFlags()
+            {
+                _flags = new JsonFlags() { HasMedia = JsonFunction.DataHasMedia(_args[0].Type._type) };
+
+                if (_args.Length > 1 && _args[1] is StringValue arg1string)
+                {
+                    _flags.IgnoreBinaryData = arg1string.Value.Contains("G");
+                    _flags.IgnoreUnsupportedTypes = arg1string.Value.Contains("I");
+                    _flags.IncludeBinaryData = arg1string.Value.Contains("B");
+                    _flags.IndentFour = arg1string.Value.Contains("4");
+                }
+
+                if (_args.Length > 1 && _args[1] is OptionSetValue arg1optionset)
+                {
+                    _flags.IgnoreBinaryData = arg1optionset.Option == "IgnoreBinaryData";
+                    _flags.IgnoreUnsupportedTypes = arg1optionset.Option == "IgnoreUnsupportedTypes";
+                    _flags.IncludeBinaryData = arg1optionset.Option == "IncludeBinaryData";
+                    _flags.IndentFour = arg1optionset.Option == "IndentFour";
+                }
+
+                _flags.Compact = !_flags.IndentFour;
+
+                if ((_flags.IncludeBinaryData && _flags.IgnoreBinaryData) ||
+                    (_flags.HasMedia && !_flags.IncludeBinaryData && !_flags.IgnoreBinaryData) ||
+                    (!_flags.IgnoreUnsupportedTypes && JsonFunction.HasUnsupportedType(_args[0].Type._type, out var _, out var _)))
+                {
+                    _flags = null;
+                    return;
+                }
+            }
+
+#pragma warning disable CS0618 // Use of [Obsolete] member: Date[Time]Value.Value
+
+            internal string JsonInternal(FormulaValue arg0, int indent)
+            {
+                return arg0 switch
+                {
+                    BlankValue blkv => "null",
+                    BooleanValue bv => bv.Value.ToString(CultureInfo.InvariantCulture).ToLowerInvariant(),
+                    ColorValue cv => GetColorString(cv.Value),
+                    DateTimeValue dtv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dtv.Value, _timeZoneInfo).ToString("yyyy-MM-dd'T'HH:mm:ss.fffK", CultureInfo.InvariantCulture)),
+                    DateValue dv => WithQuotes(TimeZoneInfo.ConvertTimeToUtc(dv.Value, _timeZoneInfo).ToString("yyyy-MM-dd", CultureInfo.InvariantCulture)),
+                    DecimalValue dmv => dmv.Value.ToString(CultureInfo.InvariantCulture),
+                    GuidValue gv => WithQuotes(gv.Value.ToString("D", CultureInfo.InvariantCulture)),
+                    NumberValue nv => nv.Value.ToString(CultureInfo.InvariantCulture),
+                    OptionSetValue osv => GetOptionSetValueString(osv),

```

Comment:
>GetOptionSetValueString [](http://example.com/codeflow?start=42&length=23)

We should use the integer value (which is language-independent), not the string value (which can vary depending on the locale)